lib/paper_trail.rb

```diff
@@ -103,59 +138,54 @@ def timestamp_field=(_field_name)
       raise(E_TIMESTAMP_FIELD_CONFIG)
     end
 
-    # Sets who is responsible for any changes that occur. You would normally use
-    # this in a migration or on the console, when working with models directly.
-    # In a controller it is set automatically to the `current_user`.
+    # See PaperTrail::Request for information
     # @api public
     def whodunnit=(value)
-      paper_trail_store[:whodunnit] = value
+      ::ActiveSupport::Deprecation.warn(
+        "PaperTrail.whodunnit= is deprecated, use PaperTrail.request.whodunnit="
+      )
+      request.whodunnit = value
     end
 
-    # If nothing passed, returns who is reponsible for any changes that occur.
-    #
-    #   PaperTrail.whodunnit = "someone"
-    #   PaperTrail.whodunnit # => "someone"
-    #
-    # If value and block passed, set this value as whodunnit for the duration of the block
-    #
-    #   PaperTrail.whodunnit("me") do
-    #     puts PaperTrail.whodunnit # => "me"
-    #   end
-    #
+    # See PaperTrail::Request for information
     # @api public
     def whodunnit(value = nil)
-      if value
-        raise ArgumentError, "no block given" unless block_given?
-
-        previous_whodunnit = paper_trail_store[:whodunnit]
-        paper_trail_store[:whodunnit] = value
-
+      if value.nil?
+        ::ActiveSupport::Deprecation.warn(
+          "PaperTrail.whodunnit is deprecated, use PaperTrail.request.whodunnit"
+        )
+        request.whodunnit
+      else
+        ::ActiveSupport::Deprecation.warn(
+          "Passing a block to PaperTrail.whodunnit is deprecated, " \
+          'use PaperTrail.request(whodunnit: "John") do .. end'
+        )
+        before = request.whodunnit
+        request.whodunnit = value
         begin
           yield
         ensure
-          paper_trail_store[:whodunnit] = previous_whodunnit
+          request.whodunnit = before
         end
-      elsif paper_trail_store[:whodunnit].respond_to?(:call)
-        paper_trail_store[:whodunnit].call
-      else
-        paper_trail_store[:whodunnit]
       end
     end
 
-    # Sets any information from the controller that you want PaperTrail to
-    # store.  By default this is set automatically by a before filter.
+    # See PaperTrail::Request for information
     # @api public
     def controller_info=(value)
-      paper_trail_store[:controller_info] = value
+      ::ActiveSupport::Deprecation.warn(
+        "PaperTrail.controller_info= is deprecated, use PaperTrail.request.controller_info="
+      )
+      request.controller_info = value
     end
 
-    # Returns any information from the controller that you want
-    # PaperTrail to store.
-    #
-    # See `PaperTrail::Rails::Controller#info_for_paper_trail`.
+    # See PaperTrail::Request for information
     # @api public
-    def controller_info
-      paper_trail_store[:controller_info]
+    def controller_info(value = nil, &block)
```

Comment:
Why did you add these parameters?  They aren't used in `Request`.  The `Request` method signature is `def controller_info`.  Wouldn't you get a `ArgumentError: wrong number of arguments` if you call this?  If so, can you add test coverage for this?

Reply:
Seems like a mistake. I didn't mean to change the method signature. Good catch.